### PR TITLE
type1 quorum transactionID updated

### PR DESCRIPTION
**Pull request by ashi31**
The last characters we get in DID are : a, e, i, m, q, u, y, 4. 
Transaction IDs for Type 1 transaction are added into IPFS, to get the relative IPFS ID so that the last character matches with Quorum DIDs.

### CODE REVIEW

**Comment by ashi31** [comment]
The Old PR is removed.

**Comment by harirubix** [comment]
This works ! But I am sure we can come with a better solution.  Could you please check the multihash structure for IPFS ? That will bring an alt options.

**Comment by ashi31** [comment]
> This works ! But I am sure we can come with a better solution. Could you please check the multihash structure for IPFS ? That will bring an alt options.

Yes sure Hari, I'll check that

File: core/quorum_initiator.go [hunk 1]
```diff
@@ -1245,3 +1257,63 @@ func (c *Core) createCommitedTokensBlock(newBlock *block.Block, smartContractTok
...
 	return nil
 }
+
+func (c *Core) generateTransactionID(transType int, sc *contract.Contract) (string, string, error) {
+	hash := util.CalculateHash(sc.GetBlock(), "SHA3-256")
+	tid := util.HexToStr(hash)
+	if transType == QuorumTypeTwo {
+		return tid, "", nil
+	}
+	os.Mkdir(c.didDir+"/tempdir", 0777)
```
Comment (gklps) [inline core/quorum_initiator.go:1267]: What happens if the directory already exists?

**Comment by harirubix** [comment]
> > This works ! But I am sure we can come with a better solution. Could you please check the multihash structure for IPFS ? That will bring an alt options.
> 
> Yes sure Hari, I'll check that

Any updates ?

**Comment by ashi31** [comment]
> > > This works ! But I am sure we can come with a better solution. Could you please check the multihash structure for IPFS ? That will bring an alt options.
> > 
> > 
> > Yes sure Hari, I'll check that
> 
> Any updates ?

We can convert the Hex Transaction ID to Base32 temporarily, and use the last character of TransactionID to match with 2nd last character of quorum DID to pick quorums. That way we'll have more buckets to select quorum from.

**Comment by harirubix** [comment]
Wontfix . Issue resolved by #177

**Pull request closed without merging**